docs/src/userguide/memoryviews.rst

```diff
@@ -436,31 +436,20 @@ The flags are as follows:
 * contiguous - contiguous and direct
 * indirect_contiguous - the list of pointers is contiguous
 
-and they can be used like this::
+and they can be used like this:
 
-    from cython cimport view
-
-    # direct access in both dimensions, strided in the first dimension, contiguous in the last
-    cdef int[:, ::view.contiguous] a
-
-    # contiguous list of pointers to contiguous lists of ints
-    cdef int[::view.indirect_contiguous, ::1] b
-
-    # direct or indirect in the first dimension, direct in the second dimension
-    # strided in both dimensions
-    cdef int[::view.generic, :] c
+.. literalinclude:: ../../examples/userguide/memoryviews/memory_layout.pyx
 
 Only the first, last or the dimension following an indirect dimension may be
 specified contiguous::
 
+    from cython cimport view
+
     # INVALID
     cdef int[::view.contiguous, ::view.indirect, :] a
     cdef int[::1, ::view.indirect, :] b
 
-    # VALID
-    cdef int[::view.indirect, ::1, :] a
-    cdef int[::view.indirect, :, ::1] b
-    cdef int[::view.indirect_contiguous, ::1, :]
+.. literalinclude:: ../../examples/userguide/memoryviews/memory_layout_2.pyx
```

Comment:
Would it make sense to reverse the order of the valid and invalid examples? That way, the `cimport` would go first and not appear twice.